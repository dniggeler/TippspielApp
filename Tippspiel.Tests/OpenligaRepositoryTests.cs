using FluentAssertions;
using Ninject;
using System.Threading.Tasks;
using BhFS.Tippspiel.Utils;
using Snapshooter.Xunit;
using Tippspiel.Contracts;
using Xunit;

namespace Tippspiel.Tests
{
    [Trait("Tippspiel","Openliga Repository")]
    public class OpenligaRepositoryTests
    {
        private readonly IFussballDataRepository _repository;

        public OpenligaRepositoryTests()
        {
            StandardKernel kernel = new StandardKernel( new TippspielModule() );
            _repository = kernel.Get<IFussballDataRepository>();
        }

        [Fact(DisplayName = "Get Matches By Group")]
        public async Task GetMatchesByGroupTest()
        {
            // given
            int groupId = 3;

            // when
            var result = await _repository.GetMatchesByGroupAsync(groupId);

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Current Group")]
        public async Task GetCurrentGroupTest()
        {
            // given

            // when
            var result = await _repository.GetCurrentGroupAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get All Groups")]
        public async Task GetAllGroupsTest()
        {
            // given
            int expectedMatches = 34;

            // when
            var groups = await _repository.GetAllGroupsAsync();

            // then
            groups.Count.Should().Be(expectedMatches);
        }

        [Fact(DisplayName = "Get All Matches")]
        public async Task GetAllMatchesTest()
        {
            // given
            int expectedMatches = 306;

            // when
            var matches = await _repository.GetAllMatchesAsync();

            // then
            matches.Count.Should().Be(expectedMatches);
        }


        [Fact(DisplayName = "Get Match By Id")]
        public async Task GetMatchById()
        {
            // given
            int matchId = 66641;

            // when
            var match = await _repository.GetMatchDataAsync(matchId);

            // then
            Snapshot.Match(match);
        }

        [Fact(DisplayName = "Get Match Not Started")]
        public async Task GetMatchByIdNotYetStarted()
        {
            // given
            int matchId = 66648;

            // when
            var match = await _repository.GetMatchDataAsync(matchId);

            // then
            Snapshot.Match(match);
        }

        [Fact(DisplayName = "Get Teams")]
        public async Task GetTeamsTest()
        {
            // given
 
            // when
            var result = await _repository.GetTeamsAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Next Match")]
        public async Task GetNextMatch()
        {
            // given

            // when
            var result = await _repository.GetNextMatchAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Last Match")]
        public async Task GetLastMatch()
        {
            // given

            // when
            var result = await _repository.GetLastMatchAsync();

            // then
            Snapshot.Match(result);
        }
    }
}
