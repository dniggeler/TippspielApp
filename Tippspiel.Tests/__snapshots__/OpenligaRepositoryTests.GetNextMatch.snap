{
  "matchID": 66648,
  "group": {
    "groupOrderID": 3,
    "groupName": "3. Spieltag"
  },
  "MatchNr": 0,
  "leagueShortcut": "bl1",
  "matchDateTime": "2023-09-01T20:30:00",
  "matchDateTimeUTC": "2023-09-01T18:30:00Z",
  "matchIsFinished": false,
  "team1": {
    "teamId": 7,
    "teamName": "Borussia Dortmund",
    "shortName": "BVB",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/6/67/Borussia_Dortmund_logo.svg/560px-Borussia_Dortmund_logo.svg.png"
  },
  "team2": {
    "teamId": 199,
    "teamName": "1. FC Heidenheim 1846",
    "shortName": "Heidenheim",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/9/9d/1._FC_Heidenheim_1846.svg/830px-1._FC_Heidenheim_1846.svg.png"
  },
  "matchResults": [],
  "HomeTeamScore": 0,
  "AwayTeamScore": 0,
  "HasVerlaengerung": false,
  "ResultType": null,
  "HasStarted": false
}
